### PR TITLE
feat: add experimental support for reverse scans public preview

### PR DESCRIPTION
**Pull request by igorbernstein2**
This adds support for Scan#setReversed().

### CODE REVIEW

File: bigtable-client-core-parent/bigtable-hbase-integration-tests-common/src/test/java/com/google/cloud/bigtable/hbase/TestScan.java [hunk 1]
```diff
@@ -302,6 +309,149 @@ public void testStartEndEquals() throws IOException {
...
+    ArrayList<Put> puts = new ArrayList<>(rowsToWrite);
+
+    // Insert some columns
+    for (int rowIndex = 0; rowIndex < rowsToWrite; rowIndex++) {
+      Put put = new Put(rowKeys[rowIndex]).addColumn(COLUMN_FAMILY, qualifier, value);
+      puts.add(put);
+    }
+    table.put(puts);
+
+    Scan scan = new Scan().setReversed(true).withStartRow(rowKeys[6]).withStopRow(rowKeys[2]);
```
Comment (mutianf) [inline bigtable-client-core-parent/bigtable-hbase-integration-tests-common/src/test/java/com/google/cloud/bigtable/hbase/TestScan.java:338]: I'm a bit confused.. so the Scan start key needs to be < stop key? which is the opposite of the query?

Comment (igorbernstein2) [reply bigtable-client-core-parent/bigtable-hbase-integration-tests-common/src/test/java/com/google/cloud/bigtable/hbase/TestScan.java:338]: yep...HBase flips it around for Scans :(

File: bigtable-client-core-parent/bigtable-hbase-integration-tests-common/src/test/java/com/google/cloud/bigtable/hbase/TestScan.java [hunk 2]
```diff
@@ -302,6 +309,149 @@ public void testStartEndEquals() throws IOException {
...
+    }
+    table.put(puts);
+
+    Scan scan =
+        new Scan()
+            .setReversed(true)
+            .setFilter(
+                new MultiRowRangeFilter(
+                    Lists.newArrayList(
+                        new RowRange(rowKeys[3], false, rowKeys[4], true),
```
[the remark this answers is not in the document]
Comment (igorbernstein2) [reply bigtable-client-core-parent/bigtable-hbase-integration-tests-common/src/test/java/com/google/cloud/bigtable/hbase/TestScan.java:388]: Yep, Hbase is inconsistent between Scan & Filter :(

**Review by mutianf — APPROVED** [review]
LGTM if the behaviors are documented somewhere! :)

File: bigtable-client-core-parent/bigtable-hbase-integration-tests-common/pom.xml [hunk 3]
```diff
@@ -124,6 +124,12 @@ limitations under the License.
       <version>${junit.version}</version>
       <scope>test</scope>
     </dependency>
+
+    <dependency>
+      <groupId>com.google.truth</groupId>
+      <artifactId>truth</artifactId>
+      <version>1.1.3</version>
```
Comment (mutianf) [inline bigtable-client-core-parent/bigtable-hbase-integration-tests-common/pom.xml:131]: nit: should this be a variable like other versions?

**Review by mutianf — APPROVED** [review]

File: bigtable-client-core-parent/bigtable-hbase-integration-tests-common/pom.xml (hunk 3 above)
Comment (igorbernstein2) [reply bigtable-client-core-parent/bigtable-hbase-integration-tests-common/pom.xml:131]: we hardcode the truth version in every other module. We can extract it to a variable, but thats for another time

**Pull request merged**